File: benchmarks/fused_inc.hs
not_ b t f = b f t
exp_ a b = \ s z -> (b a s z)
odd_ n = n not_ (\ t f -> f)

a = \ s z -> (s (s (s (s (s (s (s (s z))))))))
b = \ s z -> (s (s (s (s (s (s (s (s z))))))))

main :: IO ()
main = print $ odd_ (mul_ a b) True False

File: benchmarks/fused_inc.js
var NOT = b => t => f => b(f)(t)
var EXP = a => b => b(a)
var ODD = n => n(NOT)(t => f => f)

var S = n => s => z => s(n(s)(z));

let A = S => Z => S(S(S(S(S(Z)))));
let B = S => Z => S(S(S(S(S(Z)))));

function to_num(lam_num) {
  return lam_num((x) => x + 1)(0);
}

function to_bool(lam_bool) {
  return lam_bool(true)(false);
}

console.log(to_bool(ODD(EXP(A)(B))));
